src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -6865,6 +6962,11 @@ static MonoInst*
 emit_simd_intrinsics (const char *class_ns, const char *class_name, MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsig, MonoInst **args)
 {
 	MonoInst *ins;
+#ifdef TARGET_S390X
+	/* vector facility was introduced in z13 */
+	if (!mono_hwcap_s390x_has_vec)
+		return NULL;
```

Comment:
This is OK, but I think we need the same check in `mini_class_is_simd` as well to avoid any use of vector registers pre-z13.